drivers/nuttxcan/nuttxcan.c

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2016 UAVCAN Team
+ *
+ * Distributed under the MIT License, available in the file LICENSE.
+ *
+ */
+
+#include "nuttxcan.h"
+#include <fcntl.h>
+#include <poll.h>
+#include <string.h>
+#include <unistd.h>
+#include <nuttx/can.h>
+
+/**
+ * Initializes the NuttXCAN instance.
+ */
+int nuttxcanInit(NuttXCANInstance* out_ins, const char* can_iface_name)
+{
+    const int fd = open(can_iface_name, O_RDWR | O_NONBLOCK);
+    if (fd < 0)
+    {
+        return -1;
+    }
+
+    out_ins->fd = fd;
+    return 1;
+}
+
+/**
+ * Deinitializes the NuttXCAN instance.
+ */
+int nuttxcanClose(NuttXCANInstance* ins)
+{
+    const int close_result = close(ins->fd);
```

Comment:
Considering my last comment on return codes, you could just `return close_result;`. I also suggest to invalidate the `fd` after closing, e.g. make it -1.